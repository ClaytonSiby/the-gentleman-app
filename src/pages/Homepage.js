import React from 'react';
import { Row, Col } from 'react-bootstrap';
import styles from '../assets/scss/homepage.module.scss';

const Homepage = () => (
  <Row className={`${styles.homepage} p-0 m-0`}>
    <Col xs={12} className={`${styles.banner}`}>
      <div className="text-center">
        <h1 className="my-3">The Gentleman</h1>
        <p>Character. Intelligence. Strength. Style</p>
        <p>That makes a gentleman.</p>
      </div>
    </Col>
    <Col className={`${styles.welcomeText}`}>
      <button className="my-2 btn btn-md btn-default border" type="button">Explore More</button>
      <p className="m-2">
        Explore the style, the elegant look for your special occassion. We offer
        high quility products. The gentleman is here to give you a sophisticated
        appearance and lighten up the mood of your day. Enjoy!
      </p>
    </Col>
  </Row>
);

export default Homepage;
