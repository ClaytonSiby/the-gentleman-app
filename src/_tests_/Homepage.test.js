import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Homepage from '../pages/Homepage';

describe('Homepage', () => {
  test('Renders the Homepage component successfully', () => {
    render(<Homepage />);
  });

  test('renders consistently to the DOM', () => {
    const test = renderer.create(<Homepage />).toJSON();

    expect(test).toMatchSnapshot();
  })
});
