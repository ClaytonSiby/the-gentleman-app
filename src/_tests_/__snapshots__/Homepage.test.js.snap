// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage renders consistently to the DOM 1`] = `
<div
  className="homepage p-0 m-0 row"
>
  <div
    className="banner col-12"
  >
    <div
      className="text-center"
    >
      <h1
        className="my-3"
      >
        The Gentleman
      </h1>
      <p>
        Character. Intelligence. Strength. Style
      </p>
      <p>
        That makes a gentleman.
      </p>
    </div>
  </div>
  <div
    className="welcomeText col"
  >
    <button
      className="my-2 btn btn-md btn-default border"
      type="button"
    >
      Explore More
    </button>
    <p
      className="m-2"
    >
      Explore the style, the elegant look for your special occassion. We offer high quility products. The gentleman is here to give you a sophisticated appearance and lighten up the mood of your day. Enjoy!
    </p>
  </div>
</div>
`;
